### DISCUSSION TITLE
Sysoff problem on modified firmware

### DISCUSSION THREAD

**Discussion by yfaws** [USC-ACTLab/crazyswarm#416] [Q&A]
Hi!
I am working with a setup which has  8-10 crazyflies with loco deck + uwb, crazyswarm, Ubuntu18.04 python2, melodic. 
I am also trying to take pose data with ros topic like /cf1/pose. I always retrieve "What (): Could not find stateEstimate.y in log toc!" as error, when I enable logging pose. For solving this problem, I tried to modify firmware source code for reducing number of logging variable since log variable number is 465. I though it was too much. After deleting the log_add functions of unused logging variables from firmware source code,I loaded to firmware as binary and it solved this problem. But it surprisingly caused another problem after passing 1-2 hours especially while using chooser gui. Crazyflies randomly switch sysoff mode and  even it can not be rebooted. 
When I see the error on console print, it is "SYS: Assert failed .//src/modules/src/crtp.c:185". It looks problem is crtp packet drop, but I can not understand the actual reason because i think removing logging variable should not cause such an error.

**Comment by whoenig** [comment]
It should not be required to reduce the number of logging variables. The Crazyswarm can handle the full 465. The log toc issue is usually caused by an invalid cache: Delete `~/.ros/param*.csv` to clean you param toc cache.

Your second issue is actually a crash in the firmware. This is currently triggered when the Crazyswarm sends too many packets. The current workaround is to comment out the line (crtp.c:185) in the firmware, rebuild, and reflash. We are working on a more permanent fix by switching to a different communication backend that does not trigger this error.

**Reply by zhoukeheng** [reply]
@yfaws 
I meet the same problem as you. You just go to the stablelize.c modules in firmware and modify the name of y in the stateEstimate log section to other names (About line 657 to 670). This problem can be solved.

**Reply by yfaws** [reply]
Thanks,
To be sure if there is no remaining log and parameter cache file I used following lines in command line
find <csw_path> -name "log*.csv" -exec rm -rf "{}" ;
find <csw_path> -name "param*.csv" -exec rm -rf "{}" ;
find ~/.ros -name "cache" -exec rm -rf "{}" ;
find ~/.ros -name "log*.csv" -exec rm -rf "{}" ;
find ~/.ros -name "param*.csv" -exec rm -rf "{}" ;
then restarted all cfs,
it solved my "log toc" problem.
